### PR TITLE
Improves Rusty Hog and fixes #5730

### LINKED ISSUE #5730: Improve Rustyhog parser
Improve the Rusty Hog parser according to the comment from @damiencarol which was done after the PR was merged. https://github.com/DefectDojo/django-DefectDojo/pull/5614 
This is a followup task

### PR DESCRIPTION
**Pull request by manuel-sommer**
Improves Rusty Hog and fixes #5730

### CODE REVIEW

**Comment by damiencarol** [comment]
@manuel-sommer use the feature of "draft" of GitHub. https://github.blog/2019-02-14-introducing-draft-pull-requests/

**Comment by manuel-sommer** [comment]
@damiencarol , could you review it?

File: dojo/tools/rusty_hog/parser.py [hunk 1]
```diff
@@ -19,14 +19,7 @@ def get_findings(self, json_output, test):
...
     def parse_json(self, json_output):
-        try:
-            data = json_output.read()
-            try:
-                tree = json.loads(str(data, 'utf-8'))
-            except:
-                tree = json.loads(data)
-        except:
-            raise Exception("Invalid format")
+        tree = json.loads(json_output.read())
```
Comment (damiencarol) [inline dojo/tools/rusty_hog/parser.py:22]:
```suggestion
        tree = json.load(json_output)
```

Comment (manuel-sommer) [reply dojo/tools/rusty_hog/parser.py:22]: I have tested it without ".read()", but it throws exceptions:

uwsgi_1         |   File "/app/./dojo/tools/rusty_hog/parser.py", line 22, in parse_json
uwsgi_1         |     tree = json.loads(json_output)
uwsgi_1         |   File "/usr/local/lib/python3.8/json/__init__.py", line 341, in loads
uwsgi_1         |     raise TypeError(f'the JSON object must be str, bytes or bytearray, '
uwsgi_1         | TypeError: the JSON object must be str, bytes or bytearray, not TemporaryUploadedFile

Thus, I would keep it this way.

Comment (damiencarol) [reply dojo/tools/rusty_hog/parser.py:22]: No you must use `json.load(json_output)` format (no `s` wich mean string).
This calls a native implementation that is aware of the UTF8/16/32 situation.
https://github.com/python/cpython/blob/a6a088548063226233c08b8d35dde130746fdd10/Lib/json/__init__.py#L333-L341

**Review by damiencarol — CHANGES_REQUESTED** [review]

File: dojo/tools/rusty_hog/parser.py (hunk 1 above)
Comment (manuel-sommer) [reply dojo/tools/rusty_hog/parser.py]: Ah, nice catch.

**Review by damiencarol — APPROVED** [review]

**Review by StefanFl — APPROVED** [review]

**Pull request merged**